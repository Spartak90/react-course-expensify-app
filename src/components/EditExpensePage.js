import React from 'react';
import { connect } from 'react-redux';
import { editExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class EditExpensePage extends React.Component {
  onSubmit = () => {
    this.props.editExpense(this.props.match.params.id, this.props.expense);
    this.props.history.push('/');
  }

  render () {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit} />
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    expense: state.expenses.find((expense) => {
      return expense.id === ownProps.match.params.id;
    })
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    editExpense: (id, expense) => dispatch(editExpense(id, expense))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
