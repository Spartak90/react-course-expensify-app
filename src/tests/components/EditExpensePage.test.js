import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';

const updates = {
    amount: 50,
    createdAt: 53223525,
    description: 'testing description',
    id: 'sgdksjafa',
    note: 'sldgjalkd'
};

let id = updates.id, editExpense, history, wrapper, match;

beforeEach(() => {
    editExpense = jest.fn();
    
    history = {
        push: jest.fn()
    };

    match = {
        params: {
            id
        }
    };

    wrapper = shallow(
        <EditExpensePage
            expense={updates}
            editExpense={editExpense}
            history={history}
            match={match}
        />
    );
});

test('should render EditExpensePage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should submit callback correctly', () => {
    const ExpenseForm = wrapper.find('Connect(withRouter(ExpenseForm))');
    const onSubmit = ExpenseForm.prop('onSubmit');

    onSubmit(id, updates);

    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(id, updates);
});
