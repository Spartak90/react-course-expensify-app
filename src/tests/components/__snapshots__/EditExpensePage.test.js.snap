// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage correctly 1`] = `
<div>
  <Connect(withRouter(ExpenseForm))
    expense={
      Object {
        "amount": 50,
        "createdAt": 53223525,
        "description": "testing description",
        "id": "sgdksjafa",
        "note": "sldgjalkd",
      }
    }
    onSubmit={[Function]}
  />
</div>
`;
